src/query/storages/stream/src/stream_table.rs

```diff
@@ -332,6 +333,41 @@ impl Table for StreamTable {
         ]
     }
 
+    #[async_backtrace::framed]
+    async fn get_stream_mode(&self, ctx: Arc<dyn TableContext>) -> Result<StreamMode> {
+        match self.mode {
+            StreamMode::AppendOnly => Ok(StreamMode::AppendOnly),
+            StreamMode::Standard => {
+                let table = self.source_table(ctx.get_default_catalog()?).await?;
+                let fuse_table = FuseTable::try_from_table(table.as_ref())?;
+                let latest_segments =
+                    if let Some(snapshot) = fuse_table.read_table_snapshot().await? {
+                        HashSet::from_iter(snapshot.segments.clone())
```

Comment:
can we avoid this `clone` operation? (and the subsequent `base_snapshot.segments.clone()`)